hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestApacheClientConnectionPool.java

```diff
@@ -185,4 +232,21 @@ public void testKeepAliveCacheRemoveStaleConnection() throws Exception {
       }
     }
   }
+
+  @Test
+  public void testKeepAliveCacheClosed() throws Exception {
+    KeepAliveCache keepAliveCache = Mockito.spy(new KeepAliveCache(
+        new AbfsConfiguration(new Configuration(), EMPTY_STRING)));
+    keepAliveCache.put(Mockito.mock(HttpClientConnection.class));
+    keepAliveCache.close();
+    IOException ex = intercept(IOException.class,
```

Comment:
intercept proposed new ClosedIOException; no need to check text, but if you want to, do it in the intercept() call itself as the second argument. If there's a mismatch the original exception is rethrown, which is far more informative

Reply:
Good point, taken.